src/PSRule.Rules.Azure/rules/Azure.AKS.Rule.ps1

```diff
@@ -162,6 +182,7 @@ function global:GetAgentPoolProfiles {
                 type = $TargetObject.properties.type
                 maxPods = $TargetObject.properties.maxPods
                 orchestratorVersion = $TargetObject.properties.orchestratorVersion
+                enableAutoScaling = $TargetObject.properties.enableAutoScaling
```

Comment:
I think this will also need to be inserted after line 175 `orchestratorVersion = $_.properties.orchestratorVersion` above.

Doesn't look like there is test coverage for this but the scenario where the agentPool is a nested sub-resource in the cluster i.e. In the `"resources": [ ]` of the cluster should ideally work as well.

Reply:
This was another thing I was unsure about. If you have sub resources that are invalid, the rule record still outcome shows up at `Pass`.

Reply:
Yeah so tested this with this template: 

```json
    "resources": [
        {
            "apiVersion": "2021-02-01",
            "dependsOn": [
                "[concat('Microsoft.Resources/deployments/', 'WorkspaceDeployment-20210814190241')]"
            ],
            "type": "Microsoft.ContainerService/managedClusters",
            "location": "[parameters('location')]",
            "name": "[parameters('resourceName')]",
            "properties": {
                "kubernetesVersion": "[parameters('kubernetesVersion')]",
                "enableRBAC": "[parameters('enableRBAC')]",
                "dnsPrefix": "[parameters('dnsPrefix')]",
                "agentPoolProfiles": [
                    {
                        "name": "agentpool1",
                        "osDiskSizeGB": "[parameters('osDiskSizeGB')]",
                        "count": 3,
                        "enableAutoScaling": true,
                        "minCount": 1,
                        "maxCount": 5,
                        "vmSize": "Standard_DS2_v2",
                        "osType": "Linux",
                        "storageProfile": "ManagedDisks",
                        "type": "VirtualMachineScaleSets",
                        "mode": "System",
                        "maxPods": 110,
                        "availabilityZones": null
                    }
                ],
                "networkProfile": {
                    "loadBalancerSku": "standard",
                    "networkPlugin": "[parameters('networkPlugin')]"
                },
                "apiServerAccessProfile": {
                    "enablePrivateCluster": "[parameters('enablePrivateCluster')]"
                },
                "addonProfiles": {
                    "httpApplicationRouting": {
                        "enabled": "[parameters('enableHttpApplicationRouting')]"
                    },
                    "azurepolicy": {
                        "enabled": "[parameters('enableAzurePolicy')]"
                    },
                    "omsAgent": {
                        "enabled": false
                    }
                }
            },
            "tags": {},
            "identity": {
                "type": "SystemAssigned"
            },
            "resources": [
                {
                    "type": "Microsoft.ContainerService/managedClusters/agentPools",
                    "apiVersion": "2021-02-01",
                    "name": "[concat(parameters('resourceName'), '/agentpool2')]",
                    "dependsOn": [
                        "[resourceId('Microsoft.ContainerService/managedClusters', parameters('resourceName'))]"
                    ],
                    "properties": {
                        "osDiskSizeGB": "[parameters('osDiskSizeGB')]",
                        "count": 3,
                        "enableAutoScaling": false,
                        "vmSize": "Standard_DS2_v2",
                        "osType": "Linux",
                        "storageProfile": "ManagedDisks",
                        "type": "VirtualMachineScaleSets",
                        "mode": "System",
                        "maxPods": 110,
                        "availabilityZones": null
                    }
                }
            ]
        }
    ]
```

Which gave the following:

```pwsh

    ____  _____ ____        __
   / __ \/ ___// __ \__  __/ /__
  / /_/ /\__ \/ /_/ / / / / / _ \
 / ____/___/ / _, _/ /_/ / /  __/
/_/    /____/_/ |_|\__,_/_/\___/

Using PSRule v1.6.0+473073bb8bfb8cc11051923545e290f916850110
Using PSRule.Rules.Azure v0.0.1

----------------------------
Explore documentation: https://microsoft.github.io/PSRule/
Contribute and find source: https://github.com/microsoft/PSRule
Report issues: https://github.com/microsoft/PSRule/issues
PSRule.Rules.Azure: https://aka.ms/ps-rule-azure
----------------------------

> akstestcluster : Microsoft.ContainerService/managedClusters [1/1]

   PASS  Azure.AKS.AutoScaling

Rules processed: 1, failed: 0, errored: 0
```

Seems like I need to check how these sub resources are being checked since this doesn't look right.

Reply:
I guess another challenge here is that I'm unable to debug this by adding breakpoints inside the rules. I usually add this VSCODE configuration to debug PowerShell interactively:

```json
{
    "name": "PowerShell: Interactive Session",
    "type": "PowerShell",
    "request": "launch",
    "cwd": ""
}
```

But doesn't seem to have any effect if I place breakpoints inside rules.

Reply:
I guess another issue I'm having is that I can't set breakpoints within the rules and step through with the VSCode debugger. I ussually debug these issues with Interactive sessions but for some reason when I run `Assert-PSRule` it doesn't pick them up.

Reply:
Your example template looks correct to me.

I think the errors is related to line 173, `type = $_.type` which should be `type = $_.properties.type`.
This appears to be a bug with the existing code.

Because the node pools is not matching the scale set type it's being passed.

Reply:
Ah good catch, that went past me. Should be fixed now.